tests/test_unit/test_process.py

```diff
@@ -99,6 +120,51 @@
             'Here\'s a diagram: <a id="foo" href="../badge.png" class="bar">example</a>',
             id='html-anchor',
         ),
+        # Adverarial tests: contains >, multiple tag in line.
+        pytest.param(
+            '<img id="foo" attr="3>2" src="assets/diagram.png" alt="diagram" class="bar" /><img id="foo" attr="3>2" src="assets/diagram.png" alt="diagram" class="bar" />',
+            'README',
+            'docs/home.md',
+            '<img id="foo" attr="3>2" src="../assets/diagram.png" alt="diagram" class="bar" /><img id="foo" attr="3>2" src="../assets/diagram.png" alt="diagram" class="bar" />',
+            id='html-image-adverarial-test',
+        ),
+        pytest.param(
+            '<a id="foo" attr="3>2" href="badge.png" class="bar">foo</a><a id="foo" attr="3>2" href="badge.png" class="bar">bar</a>',
+            'README',
+            'docs/home.md',
+            '<a id="foo" attr="3>2" href="../badge.png" class="bar">foo</a><a id="foo" attr="3>2" href="../badge.png" class="bar">bar</a>',
+            id='html-anchor-adverarial-test',
+        ),
+        # Adversarial test: img no end slash
+        pytest.param(
+            'Here\'s a diagram: <img id="foo" src="assets/diagram.png" alt="diagram" class="bar">',
+            'README',
+            'docs/home.md',
+            'Here\'s a diagram: <img id="foo" src="../assets/diagram.png" alt="diagram" class="bar">',
+            id='html-image-no-end-slash',
+        ),
+        # external link
+        pytest.param(
+            '<img id="foo" attr="3>2" src="https://example.com/image.png" class="bar" />',
+            'README',
+            'docs/home.md',
+            '<img id="foo" attr="3>2" src="https://example.com/image.png" class="bar" />',
+            id='html-image-external-link',
+        ),
+        pytest.param(
+            '<a id="foo" attr="3>2" href="https://example.com/index.html" class="bar" />',
+            'README',
+            'docs/home.md',
+            '<a id="foo" attr="3>2" href="https://example.com/index.html" class="bar" />',
+            id='html-anchor-external-link',
+        ),
+        pytest.param(
+            '<a id="foo" attr="3>2" href="https://example.com" class="bar" />',
+            'README',
+            'docs/home.md',
+            '<a id="foo" attr="3>2" href="https://example.com" class="bar" />',
+            id='html-anchor-external-top-level-link',
```

Comment:
Would you mind to add a test with single quote values like `<a img='...' />`?

Reply:
Updated the regex and tests.